activerecord/test/cases/associations/eager_test.rb

```diff
@@ -1419,6 +1419,24 @@ def test_eager_load_multiple_associations_with_references
     assert david.readonly_comments.first.readonly?
   end
 
+  test "eager-loading non-readonly association" do
+    # has-one
+    firm = Firm.where(id: "1").eager_load(:account).first!
+    assert_not firm.account.readonly?
+
+    # has_and_belongs_to_many
+    project = Project.where(id: "2").eager_load(:developers).first!
+    assert_not project.developers.first.readonly?
+
+    # has-many :through
```

Comment:
Fixed and squashed in https://github.com/k0kubun/rails/commit/ffd4b4bfb115a950cece40527fbed31ed39d8df0